usr/sbin/p11sak/p11sak.c

```diff
@@ -961,14 +961,32 @@ static CK_RV set_custom_attrs_from_string(char *attr_char, CK_ATTRIBUTE *attr, C
         if (strncasecmp((attr_char + i), "false", 5) == 0) {
             attr[*count].pValue = &ckb_false;
             i += 5;
+            attr[*count].ulValueLen = sizeof(CK_BBOOL);
         } else if (strncasecmp((attr_char + i), "true", 4) == 0) {
             attr[*count].pValue = &ckb_true;
             i += 4;
+            attr[*count].ulValueLen = sizeof(CK_BBOOL);
         } else {
-            return CKR_ATTRIBUTE_VALUE_INVALID;
-        }
-        attr[*count].ulValueLen = sizeof(CK_BBOOL);
+            // if not boolean, a CK_ULONG value is assumed
+            radix = 16;
+            id = 0;
+            for (j = 0; attr_char[(i + j)] != ',' && attr_char[(i + j)] != '\0'; j++) {
+                if (j == 1 && attr_char[i + j] != 'x' && attr_char[i + j] != 'X')
+                    radix = 10;
+                substr[j] = attr_char[i + j];
+            }
+            substr[j] = '\0';
+            id = (int)strtoul(substr, NULL, radix);
+
+            attr[*count].pValue = malloc(sizeof(CK_ULONG));
```

Comment:
check if malloc failed and returned NULL is missing here.

Reply:
Done